lib/splitclient-rb/cache/routers/impression_router.rb

```diff
@@ -0,0 +1,48 @@
+module SplitIoClient
+  class ImpressionRouter
+    attr_reader :router_thread
+
+    def initialize(config)
+      @config = config
+      @listener = config.impression_listener
+      @queue = Queue.new
+      router_thread
+
+      if defined?(PhusionPassenger)
+        PhusionPassenger.on_event(:starting_worker_process) do |forked|
```

Comment:
No, for unicorn and puma we apply the fix from the Rails app side, there's also a ticket to switch to this approach for passenger: https://splitio.atlassian.net/browse/SPLIT-2604